feat: ViewingPosts-User-IsNot-Logged

NikitaVlas opened the pull request:
- show header at post if user authorized
- show button profile settings if user authorized
- redirect from main page if post kliked

NikitaVlas committed:
feat: ViewingPosts-User-IsNot-Logged

- show header at post if user authorized
- show button profile settings if user authorized
- redirect from main page if post kliked

PetrokasArtur commented on shared/ui/userFeedCard/ui/UserCardList/UserCardList.tsx:
```diff
@@ -15,17 +16,21 @@ export const UserCardList = ({ items }: Props) => {
-        />
-      ))}
+      {items.map(p => {
+        const href = `${PATH.profile}/${p.user.userId}?postId=${p.id}`
```
текущий путь стоит вынести в отдельную функцию в path, чтобы не дублировать в дальнейшем

NikitaVlas committed:
fix: IN-233reusable component UserHeader by addPublicationModalComponent

NikitaVlas committed:
fix: delete uslose component

PetrokasArtur commented on entities/newPublication/ui/AddPublicationModalComponent.tsx:
```diff
@@ -30,6 +32,11 @@ export const AddPublicationModalComponent = ({
+  const currentUserName = extractUserName(me) ?? 'User'
+
   const urlsToFiles = async (urls: string[]) => {
     const limited = urls.slice(0, 10) // сервер — до 10 изображений
```
и здесь коментарий можно удалить

PetrokasArtur commented on entities/newPublication/ui/AddPublicationModalComponent.tsx:
```diff
@@ -30,6 +32,11 @@ export const AddPublicationModalComponent = ({
   const [description, setDescription] = useState('')
   const [createPost] = useCreatePostMutation()
 
+  // берём текущего юзера (если авторизован — будет объект; иначе undefined)
```
удалить комментарий поясняет очевидное

NikitaVlas committed:
fix: delete commits and universal funk for path

NikitaVlas committed:
feat: add email by modal logout

NikitaVlas committed:
feat: sidebar refactoring

- Единый «источник правды» для меню
- Меньше лишних ре-рендеров из-за новых JSX-объектов
- Переиспользуемость и тестируемость

NikitaVlas committed:
Merge branch 'development' of https://github.com/IronDragons-best/inctagram into development

it-incubator-jenkins commented:
Can one of the admins verify this patch?

NikitaVlas committed:
Merge branch 'development' into nv/IN-280-ViewingPosts-User-IsNot-Logged

NikitaVlas committed:
fix ***
